controlpanel/frontend/static/javascripts/modules/role-names.js

```diff
@@ -0,0 +1,53 @@
+moj.Modules.roleNames = {
+  id: "role_name",
+  selectId: "role_name-select",
+  formClass: "appRoles",
+  roleListEndpoint: "/parameters/form/role-list.js",
+  roles: null,
+
+  init() {
+    if (document.querySelectorAll("." + this.formClass).length) {
+      this.getRoles().done(() => {
+        this.loadRolesToSelect();
+        this.bindEvents();
+        console.log(this.roles);
+      });
+    }
+    this.$repoSlugTypeahead = $('#' + this.id)
+  },
+
+  bindEvents() {
+    $('input[name=app_type]').change(() => {
+      this.loadRolesToSelect();
+    })
+  },
+
+  getRoles() {
+    return $.get(
+      this.roleListEndpoint,
+      (data) => {
+        this.roles = data;
+      }
+    )
+  },
+
+  loadRolesToSelect() {
+    $("#" + this.selectId).find('option:not([value=""])').remove();
```

Comment:
If you set `selectId` to `#role_name-select`, you don't need to do this concatenation.